msteams-platform/toolkit/debug-teams-app-visual-studio.md

```diff
@@ -0,0 +1,142 @@
+
+## Key features of Teams Toolkit
+
+### Prepare Teams app dependencies
+
+Teams Toolkit will prepare local debug dependencies and register a Teams app in the tenant, which your account belongs to.
+For Bot and Message Extension apps, Teams Toolkit will register and configure a bot.
+
+### Start debugging
+
+You can perform single operation, press **F5** to start debugging. The Teams Toolkit starts to build code, starts services, and launches browser.
+
+### Toggle breakpoints
+
+You can toggle breakpoints on the source codes of tabs, bots, message extensions, and Azure Functions. The breakpoints execute when you interact with the Teams app in a web browser.
+The following image shows the toggle breakpoints:
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-togglebreakpoints.png" alt-text="localdebug togglebreakpoints":::
+
+### Hot reload
+
+You can update and save the source codes of tab, bot, message extension, and Azure Functions at the same time when you're debugging the Teams app.
+And then select the hot reload button as follow to apply your edits:
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-hotreload.png" alt-text="localdebug hotreload":::
+
+Or you can enable "Hot Reload on File Save" for auto hot reload:
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-hotreload-onSave.png" alt-text="localdebug hotreload onSave":::
+  
+   > [!Tip]
+   > You can use hotreload function of VS during debug. Learn more by visiting <https://aka.ms/teamsfx-vs-hotreload>.
+
+### Stop debugging
+
+When you complete local debug, you can stop debugging. The following image shows the stop debug action:
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-stopDebugging.png" alt-text="localdebug stopDebugging":::
+
+## Customize debug settings
+
+### Use your bot endpoint
+
+Set siteEndpoint configuration in **.fx/configs/config.local.json** to your endpoint.
+
+```
+"bot": {
+    "siteEndpoint": "https://baidu.com"
+}
+```
+
+### Add environment variables
+
+You can add environment variables to **Properties/launchSettings.json** file.
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-customEnvironmentVariables.png" alt-text="localdebug customEnvironmentVariables":::
+
+### Launch the Teams app as a web app
+
+if you want to launch the Teams app as a web app instead of running in Teams client, first remove the **'launchUrl'** in **Properties/launchSettings.json**.
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/vS-localdebug-launchWebApp.png" alt-text="localdebug launchWebApp":::
+
+Then right click on the solution, in Properties > Configuration Properties > Configuration, uncheck the 'Deploy' process and select 'OK'.
```

Comment:
again. thought we should be consistent about the style. single quote or double or bold text?

Reply:
Agreed